server/queue/queue-api/src/test/java/org/apache/james/queue/api/MailQueueContract.java

```diff
@@ -0,0 +1,286 @@
+            .mimeMessage(originalMimeMessage)
+            .build());
+
+        MailQueue.MailQueueItem mailQueueItem = getMailQueue().deQueue();
+        assertThat(asString(mailQueueItem.getMail().getMessage()))
+            .isEqualTo(asString(originalMimeMessage));
+    }
+
+    @Test
+    default void queueShouldPreserveMailAttribute() throws Exception {
+        String attributeName = "any";
+        String attributeValue = "value";
+        getMailQueue().enQueue(defaultMail()
+            .attribute(attributeName, attributeValue)
+            .build());
+
+        MailQueue.MailQueueItem mailQueueItem = getMailQueue().deQueue();
+        assertThat(mailQueueItem.getMail().getAttribute(attributeName))
+            .isEqualTo(attributeValue);
+    }
+
+    @Test
+    default void queueShouldPreserveErrorMessage() throws Exception {
+        String errorMessage = "ErrorMessage";
+        getMailQueue().enQueue(defaultMail()
+            .errorMessage(errorMessage)
+            .build());
+
+        MailQueue.MailQueueItem mailQueueItem = getMailQueue().deQueue();
+        assertThat(mailQueueItem.getMail().getErrorMessage())
+            .isEqualTo(errorMessage);
+    }
+
+    @Test
+    default void queueShouldPreserveState() throws Exception {
+        String state = "state";
+        getMailQueue().enQueue(defaultMail()
+            .state(state)
+            .build());
+
+        MailQueue.MailQueueItem mailQueueItem = getMailQueue().deQueue();
+        assertThat(mailQueueItem.getMail().getState())
+            .isEqualTo(state);
+    }
+
+    @Test
+    default void queueShouldPreserveRemoteAddress() throws Exception {
+        String remoteAddress = "remote";
+        getMailQueue().enQueue(defaultMail()
+            .remoteAddr(remoteAddress)
+            .build());
+
+        MailQueue.MailQueueItem mailQueueItem = getMailQueue().deQueue();
+        assertThat(mailQueueItem.getMail().getRemoteAddr())
+            .isEqualTo(remoteAddress);
+    }
+
+    @Test
+    default void queueShouldPreserveRemoteHost() throws Exception {
+        String remoteHost = "remote";
+        getMailQueue().enQueue(defaultMail()
+            .remoteHost(remoteHost)
+            .build());
+
+        MailQueue.MailQueueItem mailQueueItem = getMailQueue().deQueue();
+        assertThat(mailQueueItem.getMail().getRemoteHost())
+            .isEqualTo(remoteHost);
+    }
+
+    @Test
+    default void queueShouldPreserveLastUpdated() throws Exception {
+        Date lastUpdated = new Date();
+        getMailQueue().enQueue(defaultMail()
+            .lastUpdated(lastUpdated)
+            .build());
+
+        MailQueue.MailQueueItem mailQueueItem = getMailQueue().deQueue();
+        assertThat(mailQueueItem.getMail().getLastUpdated())
+            .isEqualTo(lastUpdated);
+    }
+
+    @Test
+    default void queueShouldPreserveName() throws Exception {
+        getMailQueue().enQueue(defaultMail()
+            .name("name")
```

Comment:
extract to variable ?

Reply:
When I do extract I'm told "that's readable"

When I do not extract, then people ask me this.

A bit depressing. I'll do it only if you really urge me for it.